activejob/test/integration/queuing_test.rb

```diff
@@ -56,13 +56,16 @@ class QueuingTest < ActiveSupport::TestCase
     end
   end
 
-  test 'should supply a provider_job_id when available' do
-    skip unless adapter_is?(:sidekiq) || adapter_is?(:que) || adapter_is?(:delayed_job)
+  test 'should supply a provider_job_id when available for immediate jobs' do
+    skip unless adapter_is?(:delayed_job, :sidekiq, :qu, :que)
     test_job = TestJob.perform_later @id
-    refute test_job.provider_job_id.nil?, "Provider job id should be set by provider"
+    refute test_job.provider_job_id.nil?, 'Provider job id should be set by provider'
+  end
 
+  test 'should supply a provider_job_id when available for delayed jobs' do
+    skip unless adapter_is?(:delayed_job, :sidekiq, :que)
```

Comment:
I think you forgot to add the provider_job_id for `qu` in the enqueed_at method.